packages/syft/proto/core/tensor/share_tensor.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+
+package syft.core.tensor;
+
+import "proto/core/tensor/tensor.proto";
+import "proto/lib/numpy/array.proto";
+
+message ShareTensor {
+    oneof data {
+        Tensor tensor = 1;
+        syft.lib.numpy.NumpyProto array = 2;
+    };
+
+   uint32 rank = 3;
```

Comment:
Q: Should we also have ring_size in ShareTensor?

Reply:
I think for the moment this won't matter -- this is a branch that diverged a lot from "dev" and when we go back we would have that field (from what I remember we have it in dev)